datafusion/datasource/Cargo.toml

```diff
@@ -69,6 +71,7 @@ xz2 = { version = "0.1", optional = true, features = ["static"] }
 zstd = { version = "0.13", optional = true, default-features = false }
 
 [dev-dependencies]
+datafusion = { workspace = true, default-features = true }
```

Comment:
Typically to avoid this type of dependnecy there are two options:
1. Rewrite the tests to not use stuff (like SessionContext) that are in the core
2. Move the relevant tests somewhere into the core tests -- like `datafusion/core/tests/sql_integration.rs`